### PR TITLE
ARTEMIS-2085 - Improve validation of MDB activation config properties values

### PR DESCRIPTION
**Pull request by rpelisse**
Issue: https://issues.jboss.org/browse/JBEAP-14106
Upstream Issue: Upstream Issue: https://issues.apache.org/jira/browse/ARTEMIS-2085
Upstream Issue: https://issues.apache.org/jira/browse/ARTEMIS-2085

### CODE REVIEW

**Comment by michaelandrepearce** [comment]
Please can you create an activemq jira and update all references to that,not to a commercial vendors internal bugs system

**Comment by rpelisse** [comment]
Of course! I just did not had an account on Apache. I guess I will just have to create one :)

**Comment by jbertram** [comment]
INFO level message and above need to be "methodized."  See `org.apache.activemq.artemis.ra.ActiveMQRALogger`.

**Comment by rpelisse** [comment]
@michaelandrepearce there is no unit test in the artemis-ra, do you want me to add a first unit to validate this behavior? (I'm fine doing so, I just want to be sure it is what you meant).

**Comment by rpelisse** [comment]
@jbertram logger methodize - let me know if you want me to change something there.

**Comment by michaelandrepearce** [comment]
@rpelisse theres a suite of tests i believe under (org.apache.activemq.artemis.tests.integration.ra ) package in the integration test module, this would probably where i would look to add such test.

**Comment by rpelisse** [comment]
@michaelandrepearce maybe it's me, but it feels a bit like an overkill to add such a test into the integration test. I would rather move the validation logic into an Utils class (with static method) and add a simple JUnit test, with some mock object if needed. This way the validation logic can tested, enhance and modified completely out of the container. Would you mind if I did just that?

**Comment by rpelisse** [comment]
@michaelandrepearce I've updated this branch with my Unit tests. I did a bit of refactor also, I hope it's fine with you and let me know if you want me to change something!

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivation.java [hunk 1]
```diff
@@ -556,7 +556,7 @@ protected void setupDestination() throws Exception {
                   calculatedDestinationName = spec.getQueuePrefix() + calculatedDestinationName;
                }
 
-               logger.debug("Unable to retrieve " + destinationName + " from JNDI. Creating a new " + destinationType.getName() + " named " + calculatedDestinationName + " to be used by the MDB.");
+               logger.warn("Unable to retrieve " + destinationName + " from JNDI. Creating a new " + destinationType.getName() + " named " + calculatedDestinationName + " to be used by the MDB.");
```
Comment (michaelandrepearce) [inline artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivation.java]: if this is no longer debug, it requires to user proper logger, with a defined logger code.

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java [hunk 2]
```diff
@@ -603,7 +596,11 @@ public void setMaxSession(final Integer value) {
          logger.trace("setMaxSession(" + value + ")");
       }
 
-      maxSession = value;
+      if ( value < 1 ) {
+         logger.warn("Invalid number of session (negative):" + value + ", defaulting to 1.");
```
Comment (michaelandrepearce) [inline artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java]: this requires a declared  logger in ActiveMQRALogger and logger code

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationValidationUtils.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+package org.apache.activemq.artemis.ra.inflow;
```
Comment (michaelandrepearce) [inline artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationValidationUtils.java]: RAT failure, requires license headers

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationValidationUtils.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import javax.jms.Session;
+import javax.jms.Topic;
+import javax.resource.spi.InvalidPropertyException;
+
+import org.apache.activemq.artemis.ra.ActiveMQRALogger;
+
+/**
+ * A set of validation methods, isolated here to be easily testable.
+ * 
+ * @author rpelisse
```
Comment (michaelandrepearce) [inline artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationValidationUtils.java]: remove please

File: artemis-ra/src/test/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationsSpecTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
+package org.apache.activemq.artemis.ra.inflow;
```
Comment (michaelandrepearce) [inline artemis-ra/src/test/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationsSpecTest.java]: RAT failure, requires license header

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java (hunk 2 above)
Comment (rpelisse) [reply artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java]: Quite sorry about that. I forgot about those. I'll fix it up right away.

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationValidationUtils.java (hunk 3 above)
Comment (rpelisse) [reply artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationValidationUtils.java]: Same, my bad. I knew I needed to double check the license headers and forgot to do it. I'll update the PR accordingly.

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationValidationUtils.java (hunk 4 above)
Comment (rpelisse) [reply artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationValidationUtils.java]: Done.

**Comment by rpelisse** [comment]
@michaelandrepearce ok all fix now, let me know if you want me to change anything else!

**Comment by rpelisse** [comment]
retest this please

**Comment by jbertram** [comment]
@rpelisse, to restart the Travis CI build you'll need to do something like `git commit --amend` and then `git push -f <prBranch>`.

**Comment by rpelisse** [comment]
@jbertram thanks!

**Comment by jbertram** [comment]
I merged this on master and 2.6.x. @rpelisse, can you please close this?

**Comment by rpelisse** [comment]
> I merged this on master and 2.6.x. @rpelisse, can you please close this?

Gladly!

**Pull request closed without merging**